include/maidsafe/crux/detail/header.hpp

```diff
@@ -55,10 +55,10 @@ struct handshake {
     }
 
     void encode(detail::encoder& encoder) const {
-        encoder.put<std::uint16_t>(header::constant::type_handshake
-                                   | std::min<std::size_t>(3, retransmission_count)
-                                   | (ack ? header::constant::ack_type_cumulative
-                                          : header::constant::ack_type_none));
+        encoder.put<std::uint16_t>(
+            header::constant::type_handshake
+            | static_cast<std::uint16_t>(std::min<std::size_t>(3, retransmission_count))
```

Comment:
Does `std::min<std:uint16_t>(3, retransmission_count)` work instead of the `static_cast`?